Rename deb822-lossless to deb822-edit

jelmer opened the pull request:
This better reflects the crate's purpose as an editable parser that preserves formatting and allows modifications, not just lossless parsing.

jelmer committed:
Rename lossless modules and feature to edit

The 'lossless' naming focused on the parsing aspect, while 'edit' better
reflects the primary use case of these modules - editing files while
preserving formatting.

Changes:
- Rename lossless module to edit in debian-control, debian-copyright, and dep3
- Add new 'edit' feature, make 'lossless' an alias for backward compatibility
- Update all feature gates to use 'edit' instead of 'lossless'
- Update examples and documentation to use new module names
- Add deprecation warnings for old module and feature names

The old 'lossless' module and feature names continue to work but are
deprecated and will be removed in a future version.

michalfita commented:
Can this wait until all other PRs are merged and rejected? To reduce merge conflicts burden?

jelmer commented:
Yes, I'm not in a hurry - I've also specifically not merged this for that reason.

jelmer committed:
Rename deb822-lossless to deb822-edit

This better reflects the crate's purpose as an editable parser that
preserves formatting and allows modifications, not just lossless parsing.

jelmer committed:
Reformat with cargo fmt
